### PR TITLE
Detect ascs/ers cluster type

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Detect ASCS/ERS clusters. 
It checks if the `SAPInstance` resource appears in the `crm_mon` output of the `ha_discovery` message, even number of times (for multi sid environments).

This is the most basic and less invasive way of detecting it. There are more things we could check to see if the cluster is properly configured, but let's leave those for actual "checks".

**Sending the PR to `deregistration` branch, as I don't want to add this yet to `main` to not hijack the new checks ui potential release** 

Frontend coming next. Easy things, but I wanted to add more tests.

## How was this tested?
Automatic tests added.
I have tested in a real env as well

### CODE REVIEW

**Comment by nelsonkopliku** [comment]
Not directly in tower radar but nice job :eyes:

File: lib/trento/application/integration/discovery/payloads/cluster/cluster_discovery_payload.ex [hunk 1]
```diff
@@ -77,10 +78,28 @@ defmodule Trento.Integration.Discovery.ClusterDiscoveryPayload do
     do_detect_cluster_type(has_sap_hana_topology, has_sap_hana, has_sap_hana_controller)
   end
 
+  defp parse_cluster_type(%{"crmmon" => %{"groups" => groups}}) do
```
Comment (fabriziosestito) [inline lib/trento/application/integration/discovery/payloads/cluster/cluster_discovery_payload.ex:81]: This clause matches if the above one is not `nil`.  Maybe it is clearer if we invert the order and add `"clones" => nil` to this one, wdyt?

**Review by fabriziosestito — APPROVED** [review]
LGTM! Just a nitpick added. 

Love the simplicity.

File: lib/trento/application/integration/discovery/payloads/cluster/cluster_discovery_payload.ex (hunk 1 above)
Comment (arbulu89) [reply lib/trento/application/integration/discovery/payloads/cluster/cluster_discovery_payload.ex]: Right! It looks even better. Thank you!
Changed!

**Pull request merged**